src/cargo/util/toml/mod.rs

```diff
@@ -3470,3 +3470,19 @@ impl<'de> de::Deserialize<'de> for PathValue {
         Ok(PathValue(String::deserialize(deserializer)?.into()))
     }
 }
+
+// Based on Iterator::is_sorted from nightly std; remove in favor of that when stabilized.
+pub fn is_sorted(mut it: impl Iterator<Item = impl PartialOrd>) -> bool {
```

Comment:
nit: I think this would more belong in `toml_mut/mod.rs` since its related to toml modifications